libraries/AP_Scripting/AP_Scripting.h

```diff
@@ -44,15 +44,22 @@ class AP_Scripting
    // User parameters for inputs into scripts 
    AP_Float _user[4]; 
 
+    enum relp_type {
+        REPL_FILE = 0,
+        REPL_SERIAL = 1,
+    };
+
     struct terminal_s {
         int output_fd;
         off_t input_offset;
         bool session;
+        relp_type type;
+        AP_HAL::UARTDriver *uart; 
```

Comment:
``` c++
    struct terminal_s {
        int output_fd;
        off_t input_offset;
        AP_HAL::UARTDriver *uart; 
        relp_type type;
        bool session;
    } terminal;
```

Better off putting the pointer earlier in the struct and the bool at the end for padding reasons. You won't see a difference with our normal targets, but it will start to matter more on 64 bit targets, I've actually reclaimed a fair amount of memory from structs before on alignment so might as well not add to this.

Reply:
Done, thanks